src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -103,6 +109,102 @@ export class PatientViewPageStore {
         ))
     },{});
 
+    readonly geneticProfilesInStudy = remoteData(() => client.getAllGeneticProfilesInStudyUsingGET({
+        studyId: this.studyId
+    }), []);
+
+    private readonly mrnaRankGeneticProfileId = remoteData({
+        await: () => [
+            this.geneticProfilesInStudy
+        ],
+        invoke: () => {
+            const regex1 = /^.+rna_seq.*_zscores$/;
+            const regex2 = /^.*_zscores$/;
+            const chosenProfile: GeneticProfile = this.geneticProfilesInStudy.result.reduce((curr: GeneticProfile, next: GeneticProfile) => {

```

Comment:
i'm not sure i understand this reduce logic.  basically, we're just looking for the first item with a geneticProfileId which matches regex1 or regex2?   unless i'm mistaken (very possible!)  Array.find might be more semantic.  Sorry if I'm confused.   Also, this is real business logic here which requires domain knowledge to understand.  A comment explaining would be good.